libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt

```diff
@@ -72,124 +74,136 @@ private fun NavigationStatus.getRouteProgress(
     route?.let {
         val upcomingStepIndex = stepIndex + ONE_INDEX
 
-        val routeProgressBuilder = RouteProgress.Builder(route)
-        val legProgressBuilder = RouteLegProgress.Builder()
-        val stepProgressBuilder = RouteStepProgress.Builder()
+        var currentLegStep: LegStep? = null
+        var stepPoints: List<Point>? = null
+        var stepDistanceRemaining = 0f
+        var stepDistanceTraveled = 0f
+        var stepFractionTraveled = 0f
+        var stepDurationRemaining = 0.0
+
+        var currentLeg: RouteLeg? = null
+        var routeLegProgressDistanceRemaining: Float = 0f
+        var routeLegProgressDurationRemaining: Double = 0.0
+        var routeLegProgressFractionTraveled: Float = 0f
+        var routeLegProgressUpcomingStep: LegStep? = null
+
+        var routeProgressBannerInstructions: BannerInstructions? = null
+        var routeProgressCurrentState: RouteProgressState = RouteProgressState.INITIALIZED
+        var routeProgressUpcomingStepPoints: List<Point>? = null
+        var routeProgressDistanceRemaining: Float = 0f
+        var routeProgressDistanceTraveled: Float = 0f
+        var routeProgressDurationRemaining: Double = 0.0
+        var routeProgressFractionTraveled: Float = 0f
```

Comment:
Same as above, let's not use placeholder values for items that cannot reasonably be `null`. We should return a `null` `RouteProgress` if we cannot compute a valid update with all `non-nullable` values present.